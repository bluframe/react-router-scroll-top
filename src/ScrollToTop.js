// @flow

import React, { Fragment } from "react"
import { type Props } from "."

class ScrollToTop extends React.PureComponent<Props> {
  componentDidUpdate: (prevProps: Props) => void

  props: Props

  render() {
    const { children } = this.props

    return <Fragment>
      {children}
    </Fragment>
  }
}

export default ScrollToTop
