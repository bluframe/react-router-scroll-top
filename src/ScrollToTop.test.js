import Enzyme, { shallow } from "enzyme"
import React, { Fragment } from "react"
import Adapter from "enzyme-adapter-react-16"
import ScrollToTop from "./ScrollToTop"

Enzyme.configure({ adapter: new Adapter() })

jest.mock(".", () => ({}))

describe("ScrollToTop", () => {
  it("renders", () => {
    const wrapper = shallow(<ScrollToTop />)

    expect(wrapper).toMatchSnapshot()
  })

  it("renders children", () => {
    const wrapper = shallow(
      <ScrollToTop>
        <Fragment />
      </ScrollToTop>
    )

    expect(wrapper).toMatchSnapshot()
  })
})
