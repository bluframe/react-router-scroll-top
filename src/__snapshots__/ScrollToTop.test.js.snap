// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollToTop renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ScrollToTop />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": undefined,
    },
    "ref": null,
    "rendered": null,
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`ScrollToTop renders children 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ScrollToTop>
    <UNDEFINED />
  </ScrollToTop>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <UNDEFINED />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": Symbol(react.fragment),
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <UNDEFINED />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": Symbol(react.fragment),
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
